tests/ui/higher-ranked/higher-ranked-lifetime-error.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0308]: mismatched types
+  --> $DIR/higher-ranked-lifetime-error.rs:12:5
+   |
+LL |     assert_all::<_, &String>(id);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^ one type is more general than the other
+   |
+   = note: expected reference `&String`
+              found reference `&String`
```

Comment:
I'm not sure if I consider this an improvement over "higher ranked lifetime error".

Reply:
Yeah, the note is not all that helpful. But its actually very similar to our pre-NLL error (see output on 1.64).

I consider this more of a "get us back to where we were" than a "this is the best we can do"

Reply:
The "one type is more general than the other" note is the most helpful part of the diagnostic, and I think that's an improvement over "higher-ranked lifetime error" alone.